ktor-server/ktor-server-cio/jvmAndNix/src/io/ktor/server/cio/CIOApplicationEngine.kt

```diff
@@ -113,6 +117,46 @@ public class CIOApplicationEngine(
         }
     }
 
+    private suspend fun addHandlerForExpectedHeader(output: ByteWriteChannel, call: CIOApplicationCall) {
+        val continueResponse = "HTTP/1.1 100 Continue\r\n"
+        val expectHeaderValue = "100-continue"
+
+        val expectHeader = call.request.headers[HttpHeaders.Expect]?.lowercase()
+        var closeConnection = true
+
+        val expectedHeaderPhase = PipelinePhase("ExpectedHeaderPhase")
+        call.request.pipeline.insertPhaseBefore(ApplicationReceivePipeline.Before, expectedHeaderPhase)
+        call.request.pipeline.intercept(expectedHeaderPhase) {
+            val request = call.request
+            val version = HttpProtocolVersion.parse(request.httpVersion)
+            val alreadyRead = request.receiveChannel().totalBytesRead > 0
+            val noBody = request.headers[HttpHeaders.ContentLength]?.let { it.toInt() == 0 } ?: true
```

Comment:
`Content-Length` can be null even when there is a body. For example, in chunked requests